src/plugin-types.ts

```diff
@@ -180,18 +180,11 @@ export interface TraceAgent {
 }
 
 export interface Patch<T> {
-  file?: string;
-  versions?: string;
+  file: string;
+  versions: string;
   patch: (module: T, agent: TraceAgent) => void;
-  unpatch?: (module: T) => void;
-}
-
-export interface Intercept<T> {
-  file?: string;
-  versions?: string;
   intercept: (module: T, agent: TraceAgent) => T;
+  unpatch: (module: T) => void;
 }
 
-export type Instrumentation<T> = Patch<T>|Intercept<T>;
-
-export type Plugin = Array<Instrumentation<any>>;
+export type Plugin = Array<Partial<Patch<any>>>;
```

Comment:
Strictly speaking it might not because the type reflects the exported value of a module, which can be anything, including non-objects.

Reply:
I'm not sure if this is the case, but if all of the attributes in `Intercept` are optional, I would suggest just marking them all optional instead of using `Partial` because I think it is easier to understand.